libs/react-components/src/tedi/components/form/textarea/textarea.tsx

```diff
@@ -13,17 +13,21 @@ export interface TextAreaProps extends TextFieldProps {
 }
 
 export const TextArea = forwardRef<TextFieldForwardRef, TextAreaProps>((props, ref): JSX.Element => {
-  const { className, helper = [], characterLimit, onChange, ...rest } = props;
-  const [value, setValue] = React.useState<string>(props.value ?? '');
+  const { className, helper = [], characterLimit, onChange, value: externalValue, defaultValue, ...rest } = props;
+  const [innerValue, setInnerValue] = React.useState(defaultValue ?? '');
 
-  const handleInputChange = (inputValue: string) => {
-    setValue(inputValue);
-
-    if (onChange) {
-      onChange(inputValue);
-    }
-  };
+  const handleInputChange = React.useCallback(
+    (inputValue: string) => {
+      if (externalValue) {
+        onChange?.(inputValue);
+      } else {
+        setInnerValue(inputValue);
+      }
+    },
+    [externalValue, onChange]
+  );
```

Comment:
OnChange should be called always.
```suggestion
  const handleInputChange = React.useCallback(
    (inputValue: string) => {
      if (!externalValue && !(onChange || onChangeEvent) {
        setInnerValue(inputValue);
      }
      onChange?.(inputValue);
    },
    [externalValue, onChange]
  );
```